bytecomp/symtable.ml

```diff
@@ -102,7 +102,7 @@ let of_prim name =
         try Dll.find_primitive name
         with Not_found -> raise(Error(Unavailable_primitive name)) in
       let num = PrimMap.enter c_prim_table name in
-      Dll.synchronize_primitive num symb;
+      Option.iter (Dll.synchronize_primitive num) symb;
```

Comment:
I think you no longer report errors here. You should raise like in the above `Not_found` in case of `None`.

Reply:
`Dll.find_primitive` returns `None` when the symbol is found in a dll that has been opened "for checking" (when linking bytecode) as opposed to "for loading" (when dynlinking), so it is not an error condition.

But I agree this code needs to be cleared up as it is hard to tell what is going on right now.